maira2 support

eaidova opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Fixes # (issue)


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/optimum-intel/pr_1145). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

echarlaix commented on optimum/exporters/openvino/model_configs.py:
```diff
@@ -1488,6 +1488,7 @@ def __init__(
         float_dtype: str = "fp32",
         behavior: LlavaConfigBehavior = LlavaConfigBehavior.VISION_EMBEDDINGS,
         preprocessors: Optional[List[Any]] = None,
+        **kwargs,
```
why is it needed for MAIRA-2 models ?

eaidova replied:
maira processor has own arguments different from standard, this is needed to support inference for this case

eaidova replied:
automatic task detection in some cases select text-generation-with-past as task this is required to pass some extra arguments to config

echarlaix reviewed (commented):
LGTM, thanks @eaidova! Could you add a test before we can merge ?

eaidova commented:
@echarlaix I added test and model into supported list. Could you please one more time?

eaidova commented:
@echarlaix could you please help to rerun precommit. I updated model remote code for compatibility with python3.9

echarlaix commented:
> @echarlaix could you please help to rerun precommit. I updated model remote code for compatibility with python3.9

Done, looks like one last thing needs to be updated https://huggingface.co/katuni4ka/tiny-random-maira2/blob/main/processing_maira2.py#L347 to fix :
```
SyntaxError: non-default argument follows default argument
```

eaidova commented:
> > @echarlaix could you please help to rerun precommit. I updated model remote code for compatibility with python3.9
> 
> Done, looks like one last thing needs to be updated https://huggingface.co/katuni4ka/tiny-random-maira2/blob/main/processing_maira2.py#L347 to fix :
> 
> ```
> SyntaxError: non-default argument follows default argument
> ```

fixed
